### DISCUSSION TITLE
Use a global not-found page with multiple root layouts

### DISCUSSION THREAD

**Discussion by niclas-j** [vercel/next.js#56494] [App Router]
### Problem

Assume a folder structure with 2 root layouts:

```
app/
  (www)
     - layout.tsx ( a root layout)
     - page.tsx
     - not-found.tsx
  (admin)
     - layout.tsx (a root layout)
     - /admin
        -  page.tsx
```

The `not-found.tsx` does not fire for a non matching path like `/notexistingpath`.
This seems to be because the router cant determine on which path the (www) root layout sits. Because imagine we would put another `not-found.tsx` inside the (admin) route group. Then the router has no way to determine which not-found file should be called.

One solution step would be to use a root layout outside of the route groups and have 2 children layouts inside of each route group. That way the root not-found.tsx inside the app folder would indeed fire. But how can this be solved without going back to one root layout?

**Comment by niclas-j** [comment]
I also added an example repo: https://github.com/niclas-j/nextjs-notfound-example

**Comment by MkMonir** [comment]
@niclas-j, I'm encountering the same issue. I have two layouts: one for the home page (HomeLayout) and another for all other pages (OthersLayout). However, the 404 not-found page isn't working with this setup. If you've resolved the issue, could you please share the solution with me?
